### PR TITLE
Configure Jenkins via a related file in the repository

### PR DESCRIPTION
**Pull request by josephnoir**
Just playing around here, don't merge this ...

### CODE REVIEW

**Comment by josephnoir** [comment]
@Inet-Jenkins please test this

**Comment by josephnoir** [comment]
@Inet-Jenkins please test this

**Comment by josephnoir** [comment]
@Inet-Jenkins please test this

**Comment by josephnoir** [comment]
@Inet-Jenkins this test please

**Comment by josephnoir** [comment]
@Inet-Jenkins this test please

**Comment by josephnoir** [comment]
@Inet-Jenkins this test please

**Comment by josephnoir** [comment]
@Inet-Jenkins please test this

**Comment by josephnoir** [comment]
@Inet-Jenkins this test please

**Comment by josephnoir** [comment]
@Inet-Jenkins test this please

**Comment by josephnoir** [comment]
@Inet-Jenkins test this please

**Comment by Neverlord** [comment]
You're working on this for 3 months now. Do you have an estimate when we can switch to the new goodness?

**Comment by josephnoir** [comment]
Probably another 3 months. It would probably pay off if I concentrated on this for a few days instead of just spending a few hours every other week on this.

File: Jenkinsfile [hunk 1]
```diff
@@ -27,95 +204,34 @@ pipeline {
...
+          buildMatrix.eachWithIndex { entry, index ->
+              def (os, settings) = entry
+              settings['tools'].eachWithIndex { tool, toolIndex ->
+                  def labelExpr = "$os && $tool"
+                  def builds = settings['builds']
+                  xs["Build $os [$index:$toolIndex]"] = makeBuildStage(builds, labelExpr, settings)
+              }
           }
-          steps { msBuild() }
+          parallel xs
```
Comment (josephnoir) [inline Jenkinsfile]: 😲 That works in declarative pipelines now?

**Comment by Neverlord** [comment]
Note the `script` Block. You can embed scripting pipeline code in declarative pipelines to get the best of both worlds.

**Comment by Neverlord** [comment]
@josephnoir do you think we need any additional iterations or can we settle on this version? If so, would you mind cleaning up the first couple of commits (in particular commits such as `Damn -_-` and `!!`)?

**Comment by Neverlord** [comment]
> There is an option for that in the organization project, but I haven't tried to get it to run.

I think that's orthogonal to the Jenkinsfile itself. Merged. 🎉

**Pull request merged**